src/nrnoc/treeset.cpp

```diff
@@ -541,11 +541,11 @@ void nrn_lhs(neuron::model_sorted_token const& sorted_token, NrnThread& nt) {
     activsynapse_lhs();
 
     if (vec_sav_d) {
-        /* _nrn_save_d has only the contribution of electrode current
-           so here we transform so it only has membrane current contribution
+        /* vec_sav_d has only the contribution of electrode current
+           here we transform so it only has membrane current contribution
         */
         for (i = i1; i < i3; ++i) {
-            vec_sav_d[i] += vec_d[i];
+            vec_sav_d[i] = vec_d[i] - vec_sav_d[i];
         }
```

Comment:
> The only issue, I believe, is whether the fix, for perforamcne reasons, should made in nocmodl and NMODL.

@nrnhines: I don't have strong opinion here (due to lack of perf numbers?). Have you seen the first version with an addition better?

If there is no difference in practice, its OK to merge this.

Reply:
I have not measured and was hoping you had strong a priori judgement about this. At least if  future developers are looking at this line with ```git blame```, they will get pointed here. I will leave it as is. Fast imem is actually rarely turned on.